src/VisionModel/googlenet.jl

```diff
@@ -0,0 +1,52 @@
+using Flux
+
+export googlenet
+
+function conv_block(kernelsize::Tuple{Int64,Int64}, inplanes::Int64, outplanes::Int64; stride::Int64=1, pad::Int64=0)
+  conv_layer = []
+  push!(conv_layer, Conv(kernelsize, inplanes => outplanes, stride = stride, pad = pad))
+  push!(conv_layer, BatchNorm(outplanes, relu))
+  return conv_layer
+end
+
+function inception_block(inplanes, out_1x1, red_3x3, out_3x3, red_5x5, out_5x5, pool_proj)
+  branch1 = conv_block((1,1), inplanes, out_1x1)
+
+  branch2 = Chain(conv_block((1,1), inplanes, red_3x3),
+             conv_block((3,3), red_3x3, out_3x3; pad=1))        
+
+  branch3 = Chain(conv_block((1,1), inplanes, red_5x5),
+             conv_block((5,5), red_5x5, out_5x5; pad=1)) 
+
+  branch4 = Chain(MaxPool((3, 3), stride=1, pad=1),
+             conv_block((1,1), inplanes, pool_proj))
+ 
+  inception_layer = cat(branch1, branch2, branch3, branch4; dims=1)
```

Comment:
This should `cat` the branches, it should be a function that accepts an input, invokes the each branch on the input, then concats the output of each branch.

Reply:
Ah, yes. that's the problem.

Reply:
I tried doing so but there is ``dimension mismatch`` error. 
```
function conv_block(kernelsize::Tuple{Int64,Int64}, inplanes::Int64, outplanes::Int64; stride::Int64=1, pad::Int64=0)
  conv_layer = []
  push!(conv_layer, Conv(kernelsize, inplanes => outplanes, stride = stride, pad = pad))
  push!(conv_layer, BatchNorm(outplanes, relu))
  return conv_layer
end

function inceptionblock(inplanes, out_1x1, red_3x3, out_3x3, red_5x5, out_5x5, pool_proj, x)
  branch1 = conv_block((1,1), inplanes, out_1x1)

  branch2 = Chain(conv_block((1,1), inplanes, red_3x3),
             conv_block((3,3), red_3x3, out_3x3; pad=1))

  branch3 = Chain(conv_block((1,1), inplanes, red_5x5),
             conv_block((5,5), red_5x5, out_5x5; pad=1))

  branch4 = Chain(MaxPool((3, 3), stride=1, pad=1),
             conv_block((1,1), inplanes, pool_proj))

  inception_layer = cat(Chain(branch1...)(x), Chain(branch2[2]...)(Chain(branch2[1]...)(x)), Chain(branch3[2]...)(Chain(branch3[1]...)(x)),
                                      Chain(branch4[2]...)(Chain(branch4[1]...)(x)); dims=1)

  return inception_layer
end

function googlenet(x)
  layers = Chain(conv_block((7,7), 3, 64; stride=2, pad=3)...,
                 MaxPool((3,3), stride=2, pad=1),
                 conv_block((1,1), 64, 64)...,
                 conv_block((3,3), 64, 192; pad=1)...,
                 MaxPool((3,3), stride=2, pad=1),
                 inceptionblock(192, 64, 96, 128, 16, 32, 32, x),
                 inceptionblock(256, 128, 128, 192, 32, 96, 64, x)
```

Declaring a struct clears all the confusion.

Reply:
You don't need additional `Chain` around each branch. Also, `inception_layer` still needs to be a function.
```julia
inception_layer = x -> begin
  y1 = branch1(x)
  y2 = branch2(x)
  y3 = branch3(x)
  y4 = branch4(x)
  return cat(y1, y2, y3, y4; dims = 3)
end
```
You want to `cat` on the third dimension since that is the channels dim, and the inception module concatenates the filters (i.e. channels) together.